### ISSUE TITLE
Creation of gitRepo volume is broken in 1.8.0+

### ISSUE THREAD

**Issue by imkin** [kubernetes/kubernetes#54129]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Pods with volumeMount gitRepo volumes in 1.8.0 see the following error in kubelet logs (also in `kubectl describe pod <pod-name> logs`)
It is important to not that the git repo clone is done via ssh.

```
kubelet-wrapper[411]: error: cannot run ssh: No such file or directory
kubelet-wrapper[411]: fatal: unable to fork
kubelet-wrapper[411]: : exit status 128
```

I think what is happening is that hyperkube image now used to have `ssh` command in it and now it does not have it. So the git clone command that kubelet runs fails with the above stacktrace since it cannot find `ssh`.

**What you expected to happen**:
gitRepo is volume mounted.

**How to reproduce it (as minimally and precisely as possible)**:
Reproducible in kubernetes 1.8.0+ hyperkube images.

```
docker run -it gcr.io/google-containers/hyperkube:v1.8.2-beta.0 /bin/bash
# git clone <ssh git url>
Cloning into 'project1'...
error: cannot run ssh: No such file or directory
fatal: unable to fork
#
```

**Anything else we need to know?**:
This probably is due to missing ssh binary in the new 1.8.0 hyperkube images.
The commit https://github.com/kubernetes/kubernetes/commit/66b9ae7cd86e1f594f57a6e5d3b11ef9f09460b3 might be an issue. Will comment further if I find otherwise.
**Environment**:
- Kubernetes version (use `kubectl version`):   >=1.8.0
- Cloud provider or hardware configuration**: Not Application.
- OS (e.g. from /etc/os-release): 
hyperkube base image 
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

@kubernetes/sig-node-bugs

**Comment by resouer** [comment]
How did you installed Kubernetes may I ask?

**Comment by imkin** [comment]
@resouer 
https://github.com/vmware/ansible-coreos-kubernetes-master/
https://github.com/vmware/ansible-coreos-kubernetes-minion

**Comment by resouer** [comment]
Normally, I would suggest to install kubelet as a daemon on host instead of using hyperkube container. 

cc @luxas you may be interested in this issue.

**Comment by imkin** [comment]
@ixdy Need your help in identifying the fix here. I can send something for review. Suggestions are welcome.

**Comment by luxas** [comment]
The problem is that ssh isn't installed inside of the container. I don't know if it should be either, as the hyperkube image grows so large. If you install the openssh-client, it should work

**Comment by imkin** [comment]
@luxas Yes I did identify that as the problem if you see the desription I mentioned. I pointed to a git commit that might have caused this. 
I am not sure what package installation has changed in the hyperkube container. "ssh" used to be available and now it is not. For the gitrepo feature to work it should be available.

**Comment by luxas** [comment]
Ah okay. If it was there before, we might want to add it back. SGTM.

**Comment by ixdy** [comment]
Yeah, I switched the hyperkube image base from `debian` to our `debian-base` with some additional dependencies added back. Apparently due to lack of volume + hyperkube test coverage, I've missed a number of dependencies needed.

cc @tallclair @rphillips

**Comment by dims** [comment]
@luxas @imkin we could use dropbear as a replacement for just the ssh client?

https://gist.github.com/Schiznitz/03ece08708b822cd5e139548c09b6292
http://tumblelog.jauderho.com/post/151678345/using-dropbear-with-git

**Comment by euank** [comment]
I would prefer to use the openssh client over the dropbear client. It has wider usage and a stronger security track record IMO.

Saving a couple MB isn't worth using worse utilities, however much fun "dockerfile-golfing" is.

**Comment by ixdy** [comment]
https://github.com/kubernetes/kubernetes/pull/54250 to fix

**Comment by ixdy** [comment]
cherrypick to release-1.8: #54325

**Comment by ixdy** [comment]
This should be fixed in v1.8.3.
